tests/test_app.py

```diff
@@ -83,19 +79,30 @@ async def mock_globus_id(*args, **kwargs):
 
 class AppClient:
     @classmethod
-    async def create(cls, config, mock_username=None):
-        if mock_username is None:
-            mock_username = "testuser"
-        app = await mock_auth_app(config, mock_username)
+    async def create(cls, config: configparser.ConfigParser, auth_token: str | None, cookies: dict[str, str] | None = None) -> "AppClient":
+        """
+        Creating this with an auth token sets the headers call by default to use the
+        provided auth token in the Authorization header.
+        If you want to make a call with a different header, you can
+        do so by setting headers directly in the call.
+        """
+        app = await mock_globus_app(config, auth_token)
         server = test_utils.TestServer(app)
-        return AppClient(server)
+        return AppClient(server, auth_token, cookies)
 
-    def __init__(self, server: test_utils.TestServer):
+    def __init__(self, server: test_utils.TestServer, auth_token: str | None, cookies: dict[str, str] | None):
         self.server = server
+        self.auth_token = auth_token
+        self.cookies = cookies
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> test_utils.TestClient:
         await self.server.start_server(loop=asyncio.get_event_loop())
-        self.client = test_utils.TestClient(self.server, loop=asyncio.get_event_loop())
+        headers = {}
+        if self.auth_token is not None:
+            headers["Authorization"] = self.auth_token
+        self.client = test_utils.TestClient(
+            self.server, loop=asyncio.get_event_loop(), headers=headers, cookies=self.cookies
+        )
```

Comment:
Headers set here can be overridden during tests, these are defaults. Same goes for cookies, but they're only used in a few places to test the cookie detection code.